python/cudf/cudf/_lib/groupby.pyx

```diff
@@ -225,3 +242,35 @@ def _drop_unsupported_aggs(Table values, aggs):
         raise DataError("No numeric types to aggregate")
 
     return result
+
+
+def _is_all_scan_aggregate(aggs):
+    """
+    Returns true if all are scan aggregations.
+    Raises
+    ------
+    NotImplementedError
+        If both reduction aggregations and scan aggregations are present.
```

Comment:
Not allowing this in cudf because pandas itself produces wrong output when both scan and reduction aggregations are passed together. (mapping of index gets wrong. could be bug in pandas)
```python
In [1]: import pandas as pd


In [2]: pdf = pd.DataFrame({'a' : [2, 2, 1, 1, 6], 'b' : [2, 2, 1, 1, 3]})


In [3]: pdf.groupby('a').agg(['sum'])
Out[3]: 
    b
  sum
a    
1   2
2   4
6   3

In [4]: pdf.groupby('a').agg(['cumsum'])
Out[4]: 
       b
  cumsum
0      2
1      4
2      1
3      2
4      3

In [5]: pdf.groupby('a').agg(['cumsum', 'sum'])
Out[5]: 
       b     
  cumsum  sum
0    2.0  NaN
1    4.0  2.0
2    1.0  4.0
3    2.0  NaN
4    3.0  NaN
6    NaN  3.0
```

Reply:
👍

Reply:
I think we should raise an issue on Pandas regarding this because this behavior is weird.

Reply:
created https://github.com/pandas-dev/pandas/issues/40695

Reply:
@karthikeyann once the pandas issue is resolved and we enable mixed scan/aggregate operations, we will need to partition the creation of aggregation requests into separate lists of scan and aggregate requests so that we can call the appropriate `libcudf` function on each subset, correct? If so, would it be simpler to engineer that now, or would you prefer to wait until we reenable this? I think it's fine to wait, but assuming I understand correctly it won't be a completely trivial change since we'll have to split up the aggregations into scan and aggregate operations, farm out to `libcudf`, then recombine them in the original order, so it would be helpful to make a comment describing that somewhere.

Reply:
pandas issue should be resolved and decide the behavior. Right now, it seems, not supporting this is good idea otherwise this might become a breaking change for cudf. Based on the behavior decided, the change could be trivial or not.